.github/workflows/test.yml

```diff
@@ -5,6 +5,10 @@ on:
   push:
     branches: [main]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref_name }}
+  cancel-in-progress: true
```

Comment:
What does this do exactly? :)

Reply:
stopping other Pipeline runs for the same PR (not wasting resources)